apps/sms/test/marionette/mocks/mock_navigator_moz_mobile_message.js

```diff
@@ -32,6 +32,70 @@ Services.obs.addObserver(function(document) {
     }
   }
 
+  function generateCanvas(width, height) {
+    var canvas = document.createElement('canvas');
+    var context = canvas.getContext('2d');
+
+    canvas.width = width;
+    canvas.height = height;
+
+    var linearGradient = context.createLinearGradient(0, 0, width, height);
+    linearGradient.addColorStop(0, 'blue');
+    linearGradient.addColorStop(1, 'red');
+
+    context.fillStyle = linearGradient;
+    context.fillRect (0, 0, width, height);
+
+    return canvas;
+  }
+
+  function generateImageBlob(width, height, type, quality) {
+    var canvas = generateCanvas(width, height);
+
+    return new Promise((resolve) => {
+      canvas.toBlob((blob) => {
+        canvas.width = canvas.height = 0;
+        canvas = null;
+
+        resolve(blob);
+      }, type, quality);
+    });
+  }
+
+  function rehydrateAttachments(storage) {
+    var attachmentPromises = [];
+
+    storage.threads.forEach((thread) => {
+      thread.messages.forEach((message) => {
+        if (!message.attachments) {
+          return;
+        }
+
+        message.attachments.forEach((attachment, i) => {
+          var blobPromise;
+
+          if (attachment.type.indexOf('image') >= 0) {
+            blobPromise = generateImageBlob(
+              attachment.width, attachment.height, attachment.type
+            );
+          } else {
+            blobPromise = Promise.resolve(
+              new window.Blob([attachment.content], { type: attachment.type })
+            );
+          }
+
+          attachmentPromises.push(
+            blobPromise.then((blob) => {
+              message.attachments[i] = Cu.cloneInto({ content: blob }, window);
+            })
+          );
+        });
+      });
+    });
+
+    return Promise.all(attachmentPromises).then(() => storage);
```

Comment:
Yeah, I have a mixed feelings as well, but in the end I left it as is because ```rehydrateAttachments``` basically modifies storage and returns that modified storage, I can return empty promise there instead, and below do smth like this:
```
storagePromise = storagePromise.then((storage) => {
	return rehydrateAttachments(storage || {
	  threads: new Map(),
	  recipientToThreadId: new Map(),
	  uniqueMessageIdCounter: 0
	}).then(() => storage);
});
```
Do you like it better? :)